libV2/index.js

```diff
@@ -0,0 +1,301 @@
+
+          // this is directed graph, and hence have only one parent.
+          parent = collectionTree.node(parent && parent[0]);
+
+          // if the item construct does not exist add and initialize it to zero
+          if (!parent.ref.item) {
+            parent.ref.item = [];
+          }
+
+          // push the folder in the item that is in question
+          parent.ref.item.push(requestObject);
+
+          // set the ref for the newly created request in this.
+          collectionTree.setNode(nodeIdentified,
+            Object.assign(node, {
+              ref: _.last(parent.ref.item)
+            }));
+
+          break;
+        }
+
+        default: break;
+      }
+    });
+
+    return cb(null, {
+      result: true,
+      output: [{
+        type: 'collection',
+        data: collection
+      }],
+      analytics: this.analytics || {}
+    });
+  },
+
+  /**
+   *
+   * @description Takes in a request collection (transaction object) and validates it against
+   * corresponding definition matching endpoint
+   *
+   * @param {Object} context - Required context from related SchemaPack function
+   * @param {Array} transactions - Transactions to be validated
+   * @param {*} callback return
+   * @returns {boolean} validation
+   */
+  validateTransactionV2(context, transactions, callback) {
+    let schema = context.openapi,
+      options = context.computedOptions,
+      concreteUtils = context.concreteUtils,
+      componentsAndPaths = { concreteUtils },
+      schemaCache = context.schemaFakerCache,
+      matchedEndpoints = [];
+
+    context.schemaCache = context.schemaCache || {};
+    context.schemaFakerCache = context.schemaFakerCache || {};
+    Object.assign(componentsAndPaths, concreteUtils.getRequiredData(schema));
+
+    // create and sanitize basic spec
+    schema.servers = _.isEmpty(schema.servers) ? [{ url: '/' }] : schema.servers;
+    schema.securityDefs = _.get(schema, 'components.securitySchemes', {});
+    schema.baseUrl = _.get(schema, 'servers.0.url', '{{baseURL}}');
+    schema.baseUrlVariables = _.get(schema, 'servers.0.variables');
+
+    // Fix {scheme} and {path} vars in the URL to :scheme and :path
+    schema.baseUrl = fixPathVariablesInUrl(schema.baseUrl);
+
+    // check validity of transactions
+    try {
+      // add Ajv options to support validation of OpenAPI schema.
+      // For more details see https://ajv.js.org/#options
+      let ajv = new Ajv({
+          allErrors: true,
+          strict: false
+        }),
+        validate,
+        res;
+      addFormats(ajv);
+      validate = ajv.compile(transactionSchema);
+      res = validate(transactions);
+
+      if (!res) {
+        return callback(new OpenApiErr('Invalid syntax provided for requestList', validate.errors));
+      }
+    }
+    catch (e) {
+      return callback(new OpenApiErr('Invalid syntax provided for requestList', e));
+    }
+
+    return setTimeout(() => {
```

Comment:
@VShingala why this?

Reply:
@sujayvenaik Removing this as not needed, it was part of older code.